app/javascript/admin/controllers/creator_settings_controller.js

```diff
@@ -67,4 +67,53 @@ export class CreatorSettingsController extends Controller {
 
     reader.readAsDataURL(firstFile);
   }
+
+  /**
+   * Updates ths branding/colors on the Creator Settings Page.
+   *
+   * @param {Event} event
+   */
+  updateBranding(event) {
+    const color = event.target.value;
+
+    document.documentElement.style.setProperty('--accent-brand', color);
+
+    // responsible for the hover effect over the button.
+    // We need to recalculate this in javascript as it's currently being calculated in ruby
+    document.documentElement.style.setProperty(
+      '--accent-brand-darker',
+      this.updatedBrightness(color, 0.85),
+    );
+  }
+
+  updatedBrightness(color, amount = 1) {
+    const rgbObj = this.hexToRgb(color);
+    Object.keys(rgbObj).forEach((key) => {
+      rgbObj[key] = Math.round(rgbObj[key] * amount);
+    });
+
+    return this.rgbToHex(rgbObj['r'], rgbObj['g'], rgbObj['b']);
+  }
+
+  hexToRgb(hex) {
+    const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+    return result
+      ? {
+          r: parseInt(result[1], 16),
+          g: parseInt(result[2], 16),
+          b: parseInt(result[3], 16),
+        }
+      : null;
+  }
+
+  rgbToHex(r, g, b) {
+    return `#${this.rgbParameterToHex(r)}${this.rgbParameterToHex(
+      g,
+    )}${this.rgbParameterToHex(b)}`;
+  }
+
+  rgbParameterToHex(param) {
+    const hex = param.toString(16);
+    return hex.length == 1 ? `0${hex}` : hex;
+  }
```

Comment:
Any chance we could make these functions somehow global/shared? I'm pretty sure we would benefit from these elsewhere, like for tags coloring or for future themes customization for admins, etc.. pretty_please.gif

Reply:
I did think of it however I didn't do that because I didn't want to pre-emptively generalize that we would use that elsewhere and thought that it could be pulled out when we needed it.  

I'm happy to do it though if we think that there is a need for it. 
@nickytonline do you have a suggestion on where in the codebase a generalized function like this would live?

Reply:
No **strong** opinion here... I personally think we should do it, because it's possible no one will actually know/remember that we have this function existing here when we get to implement that elsewhere (places I've mentioned earlier). It's more likely to be discoverable when placed in shared/global place.  But that's just my $0.02 - as I said - it's not the hill to die on for me :D.

Reply:
That rationale makes sense to me :) You're right about people forgetting it exists 😁 I don't think this should hold the PR from being approved but I will simultaneously move this out into a global function. Thanks for bringing it up @ludwiczakpawel.

Reply:
I think in a case like this it's quite good to split it out into a utility purely for being able to unit test it in isolation  (which I think would be really useful for these helpers)

Reply:
@aitchiss sounds good, and a unit test for this will be useful as well. 
 
Regarding the placement of the utility file in the codebase, does this seem like a good [folder](https://github.com/forem/forem/tree/main/app/javascript/admin) to put it in alongside messageUtilities.js

Reply:
Yeah I think that probably makes most sense at the moment, since we're not anticipating (yet) these utils being used outside of admin areas 👍

Reply:
We have a utilities folder in the frontend. Consider `app/javascript/utilities/colors.js`

Reply:
Done ✅